lib/src/model/board_editor/board_editor_controller.dart

```diff
@@ -65,6 +66,56 @@ class BoardEditorController extends _$BoardEditorController {
     _updatePosition(readFen(fen).lock);
   }
 
+  /// Calculates the squares where an en passant capture could be possible.
+  SquareSet calculateEnPassantOptions() {
```

Comment:
Instead of a public method, make a private one and add an `enPassantSquares` field to the state that exposes this computation.